modules/products/bamboo/locals.tf

```diff
@@ -1,12 +1,20 @@
 locals {
   product_name = "bamboo"
+  agent_name = "bamboo-agent"
 
-  helm_chart_repository     = "https://atlassian.github.io/data-center-helm-charts"
-  bamboo_helm_chart_version = var.bamboo_configuration["helm_version"]
-  agent_helm_chart_version  = var.bamboo_agent_configuration["helm_version"]
-  number_of_agents          = var.bamboo_agent_configuration["agent_count"]
+  # Install local helm charts if local helm chart path is provided (for test purposes)
+  local_helm_charts_path = var.local_helm_charts_path == null ? "" : "${var.local_helm_charts_path}/src/main/charts"
+  use_local              = fileexists("${local.local_helm_charts_path}/${product_name}/Chart.yaml")
 
-  bamboo_software_resources = {
+  helm_chart_repository     = local.use_local ? null : "https://atlassian.github.io/data-center-helm-charts"
+  bamboo_helm_chart_name    = local.use_local ? "${local.local_helm_charts_path}/${product_name}" : local.product_name
+  bamboo_helm_chart_version = local.use_local ? null : var.bamboo_configuration["helm_version"]
+
+  agent_helm_chart_name    = local.use_local ? "${local.local_helm_charts_path}/${agent_name}" : local.agent_name
```

Comment:
We don't check for the existence of the local `bamboo-agent` chart similar to the `use_local` variable. Should we check for it?

Reply:
Here this is my design:

We introduce this feature for e2e test. So, all Helm charts either pick from remote registry or local machine. so we have just one variable to define the root folder of `data-center-helm-charts` project. This check is for making sure the provided path is related to `data-center-helm-charts` project, and as we confirm the expected file is there, here we assumed the complete project is existed including all helm charts. 

But, your point is valid, we can add another check to make sure `bamboo-agent` is there, and this will cover an unusual use-case when user cloned the helm chart project and then deleted some folders/files (including bamboo agent chart file) or made a uncompleted copy of the project just for some products including bamboo and excluding bamboo agent.  (note we add `/src/main/charts` and then add `</<product>/Chart.yaml` to the provided local folder to make sure we have the root folder of the specific project [helm chart] to proceed).

Reply:
I refactored the code and made ability to use local Helm charts individually for each chart. In this case we can define to install any selective combinations for using local or remote helm chart.